src/main/java/com/reservation/instructor/domain/Specialty.java

```diff
@@ -0,0 +1,11 @@
+package com.reservation.instructor.domain;
+
+import com.reservation.instructor.domain.exception.SpecialtyCannotBeNullException;
+
+record Specialty(String value) {
+    public Specialty {
+        if (value == null || value.isBlank()) {
+            throw new SpecialtyCannotBeNullException();
+        }
+    }
+}
```

Comment:
Tutaj też warto dołożyć limit długości. A może również walidować na podstawie formatu (regex), ale to myślę, że możemy spokojnie zostawić na później.